### PR TITLE
Fix access to carto config for embedded cluster

### PR DESCRIPTION
**Pull request by moicalcob**

### CODE REVIEW

File: manifests/kots-config.yaml [hunk 1]
```diff
@@ -191,26 +191,28 @@ spec:
...
           type: text
           default: *when_is_k8s_platform
-        - name: isK0sPlatform
+        - name: isEmbeddedCluster
           title: Embedded Cluster Platform
-          when: &when_is_k0s_platform >-
-            {{repl $result :=  (eq Distribution "k0s") -}}
+          # TODO: For the moment we're checking if it's embedded cluster if we can't obtain the k8s platform, but we should change this to a more reliable way when allowed by Replicated
+          when: &when_is_embedded_cluster >-
+            {{repl $result :=  (empty Distribution) -}}
```
Comment (ManuelLR) [inline manifests/kots-config.yaml:198]: can you change this condition to be compatible for the future?
```suggestion
            {{repl $result := or (empty Distribution) (eq Distribution "k0s") -}}
```

File: manifests/kots-config.yaml [hunk 2]
```diff
@@ -191,26 +191,28 @@ spec:
...
-            {{repl $cond5 := (ConfigOptionEquals "TLSTerminationLayer" "accessToCartoRouterHTTPS") -}}
-            {{repl $cond6 := (eq Distribution "eks") -}}
-            {{repl $result := or (and $cond1 $cond2 $cond3) (and (or $cond1 $cond6) (and $cond4 $cond5)) -}}
+            {{repl $isGke := (eq Distribution "gke") -}}
+            {{repl $isEks := (eq Distribution "eks") -}}
+            {{repl $defaultAccessModeSelected := (ConfigOptionEquals "accessToCartoModeK8s" "accessToCartoModeK8sDefault") -}}
+            {{repl $provideCustomSSLCertsSelected := (ConfigOptionEquals "accessToCartoTLSGKECertificateBehaviour" "use_custom_certificates_files_provided") -}}
+            {{repl $customAccessModeSelected := (ConfigOptionEquals "accessToCartoModeK8s" "accessToCartoModeK8sCustom") -}}
+            {{repl $terminateTLSInCARTOSelected := (ConfigOptionEquals "TLSTerminationLayer" "accessToCartoRouterHTTPS") -}}
+            {{repl $isEmbeddedCluster := (empty Distribution) -}}
```
Comment (ManuelLR) [inline manifests/kots-config.yaml:214]: I think that you should use the value calculated on the previous step, right?

File: manifests/kots-config.yaml [hunk 3]
```diff
@@ -239,20 +241,20 @@ spec:
           hidden: true
           readonly: true
           when: &when_router_https_enabled >-
-            {{repl $cond1 := (ConfigOptionEquals "accessToCartoModeK8s" "accessToCartoModeK8sCustom") -}}
-            {{repl $cond2 := (eq Distribution "k0s") -}}
-            {{repl $cond3:= (ConfigOptionEquals "TLSTerminationLayer" "accessToCartoRouterHTTPS") -}}
-            {{repl $result := and (or $cond1 $cond2) ($cond3) -}}
+            {{repl $customAccessModeSelected := (ConfigOptionEquals "accessToCartoModeK8s" "accessToCartoModeK8sCustom") -}}
+            {{repl $isEmbeddedCluster := (empty Distribution) -}}
```
Comment (ManuelLR) [inline manifests/kots-config.yaml:245]: Same as before, I think we should use the field already calculated

File: manifests/kots-config.yaml [hunk 4]
```diff
@@ -263,10 +265,10 @@ spec:
           type: text
           required: true
           default: "carto-onprem.lan"
-          validation:
-            regex:
-              pattern: ^([a-zA-Z0-9_\\-]+\\.)+[a-zA-Z]{2,}(:[0-9]+)?$
-              message: "The domain must be only a domain/subdomain. We don't support paths or protocols."
+          # validation:
```
Comment (ManuelLR) [inline manifests/kots-config.yaml:268]: I have the fix here https://github.com/CartoDB/carto-selfhosted-helm/pull/478 so we should merge it before next release

File: manifests/kots-config.yaml [hunk 5]
```diff
@@ -293,17 +295,17 @@ spec:
...
           type: label
-          when: *when_is_k0s_platform
+          when: *when_is_embedded_cluster
 
           # TLS
         - name: TLSTerminationLayer
           title: TLS Termination
           type: select_one
-          when: '{{repl or (ConfigOptionEquals "accessToCartoModeK8s" "accessToCartoModeK8sCustom") (eq Distribution "k0s") }}'
+          when: '{{repl or (ConfigOptionEquals "accessToCartoModeK8s" "accessToCartoModeK8sCustom") (empty Distribution) }}'
```
Comment (ManuelLR) [inline manifests/kots-config.yaml:308]: Same, use already calculated field

File: manifests/kots-config.yaml (hunk 1 above)
Comment (moicalcob) [reply manifests/kots-config.yaml:198]: We don't have a last word from replicated about the way they're going to let us check if it's an embedded cluster or not 🤔 I have some doubts here, I added the TODO in line 196 for that

File: chart/templates/_helpers.tpl [hunk 6]
```diff
@@ -941,12 +941,13 @@ Return the absolute path where the Google Secret will be mounted
 {{/*
 Return the proper Carto TLS Secret name
 FIXME: Deprecated in favor of router.tlsCertificates and gateway.tlsCertificates
+TODO: We have to regenerate the secret if the private key changes
 */}}
 {{- define "carto.tlsCerts.secretName" -}}
```
[the remark this answers is not in the document]
Comment (moicalcob) [reply chart/templates/_helpers.tpl:946]: Done!

File: chart/templates/_helpers.tpl [hunk 7]
```diff
@@ -941,12 +941,13 @@ Return the absolute path where the Google Secret will be mounted
...
 {{- if .Values.tlsCerts.existingSecret.name -}}
 {{- .Values.tlsCerts.existingSecret.name -}}
 {{- else -}}
```
[the remark this answers is not in the document]
Comment (moicalcob) [reply chart/templates/_helpers.tpl:950]: Done!

**Review by ManuelLR — APPROVED** [review]
LGTM ! !

**Pull request merged**